File: pyround1/119.杨辉三角-ii.py
#
# @lc app=leetcode.cn id=119 lang=python
#
# [119] 杨辉三角 II
#

# @lc code=start
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        r = [1]
        for i in range(1, rowIndex+1):
            r.insert(0,0)
            for j in range(i):
                r[j] =r[j] + r[j+1]
        return r
# @lc code=end

File: cpprounttmp/119.杨辉三角-ii.cpp
/*
 * @lc app=leetcode.cn id=119 lang=cpp
 *
 * [119] 杨辉三角 II
 */

// @lc code=start
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> numbers;
        for(int i = 0; i <= rowIndex; i++){
            numbers.push_back(vector<int>(i+1, 1));
        }
        for(int i = 0; i <= rowIndex; i++){
            for(int j = 1; j<i; j++){
                numbers[i][j] = numbers[i-1][j-1] + numbers[i-1][j];
            }
        }
        return numbers[rowIndex];

    }
};
// @lc code=end

